### ISSUE TITLE
[PDP] Clicking on "Eye" icon does not toggle visible state

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#1474]
### Expected behavior
Clicking on Eye icon should toggle visible state

### Actual Behavior
Clicking on Eye icon shows "Product is now hidden", clicking on it again shows "Product is now hidden" again and eye icon still shows "slash" through it. There is no way to make the product visible again with this control.

### Steps to Reproduce the Behavior
1. Login
1. Navigate to product grid
2. Click on Eye icon again, observe that toast message is the same.

http://g.recordit.co/SKYKPBenoV.gif

**Comment by zenweasel** [comment]
Actually the Eye icon in the sidebar has the same behavior.

**Comment by aaronjudd** [comment]
Is that because the change isn't published?

**Comment by zenweasel** [comment]
Do I need to "publish" changing it to Hidden before I can change it back to Visible?

**Comment by aaronjudd** [comment]
I think that you do, but I find this UX / workflow confusing, so I'm also a little uncertain. 
When performing the operations from the settings panel, it the "publish changes" state is different when using the "make visible" button versus the "publish" control, which is also confusing.

![xf3hjhakki](https://cloud.githubusercontent.com/assets/439959/19137599/eb1e28ae-8b29-11e6-9f42-dd0e1f85f24a.gif)

**Comment by zenweasel** [comment]
I'm thinking it's just not toggling:
```
I20161006-09:02:23.316(8)? 01:02:23.315Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:02:25.920(8)? 01:02:25.920Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:06:00.643(8)? 01:06:00.643Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:06:05.780(8)? 01:06:05.780Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:06:09.046(8)? 01:06:09.046Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:06:11.964(8)? 01:06:11.964Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:06:19.684(8)? 01:06:19.683Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:06:32.723(8)? 01:06:32.722Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:07:29.105(8)? 01:07:29.105Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:07:32.056(8)? 01:07:32.056Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:07:55.283(8)? 01:07:55.283Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:07:59.513(8)? 01:07:59.512Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:08:03.152(8)? 01:08:03.150Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:13:37.251(8)? 01:13:37.250Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:13:46.888(8)? 01:13:46.888Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:13:51.918(8)? 01:13:51.918Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
I20161006-09:13:55.848(8)? 01:13:55.847Z  INFO Reaction: toggle product visibility  BCTMZ6HTxFSppJESk false
```

**Comment by mikemurray** [comment]
@zenweasel you're right, it isn't toggling.

**Comment by mikemurray** [comment]
The eye icon in the sidebar is hooked up to different stuff for the OLD PDP page. :(

It calls a meteor method that does some legacy publish stuff, and is still around otherwise the OLD PDP page would not be able to be published.

tl;dr Yes its confusing and should be fixed
